presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -176,6 +177,13 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
     long getRetainedSizeInBytes();
 
     /**
+     * {@code consumer} visits each of the internal data container and accepts the size for it.
+     * This method can be helpful in cases such as memory counting for internal data structure.
+     * Also, the method should be non-recursive and only visit the elements at the top level.
```

Comment:
Also note that consumer should be invoked at least once with this block and the size of the instance object.